### PR TITLE
add support for /dist/foo.json to get reviews

### PR DESCRIPTION
**Pull request by leejo**
please see discussion at https://github.com/CPAN-API/metacpan-web/issues/1653

### CODE REVIEW

File: lib/CPANRatings/Control.pm [hunk 1]
```diff
@@ -62,6 +64,60 @@ sub _calc_auth_token {
...
+sub as_json {
+    my ($self, $reviews, $mode, $id) = @_;
+
+    my $data = {
+        $mode => $id,
+        reviews => [],
+    };
+
+    # arbitrary choice of one year being outdated...
+    my $outdated = DateTime->now->subtract( years => 1 );
```
Comment (abh) [inline lib/CPANRatings/Control.pm]: a slightly fancier version would be "a year ago or at the most recent release; whichever is later" or something like that.

File: lib/CPANRatings/Control.pm [hunk 2]
```diff
@@ -62,6 +64,60 @@ sub _calc_auth_token {
...
+    }
+
+    if ( my @reviews = @{ $data->{reviews} } ) {
+
+        my @recent      = grep { !$_->{outdated} } @reviews;
+        my @helpful     = grep { $_->{helpful} } @reviews;
+        my @recent_help = grep { $_->{helpful} && !$_->{outdated} } @reviews;
+
+        $data->{ratings} = {
+            all            => @reviews ? sprintf( "%.1f",$sum{all} / @reviews ) : 0,
```
Comment (abh) [inline lib/CPANRatings/Control.pm]: I think make this "all helpful"; we should just ignore the unhelpful reviews for any average rating.

File: lib/CPANRatings/Control.pm [hunk 3]
```diff
@@ -62,6 +64,60 @@ sub _calc_auth_token {
...
+            recent         => @recent ? sprintf( "%.1f",$sum{recent} / @recent ) : 0,
+            helpful        => @helpful ? sprintf( "%.1f",$sum{helpful} / @helpful ) : 0,
+            recent_helpful => @recent_help ? sprintf( "%.1f",$sum{recent_helpful} / @recent_help ) : 0,
```
Comment (abh) [inline lib/CPANRatings/Control.pm]: same change here; recent_helpful -> recent.

**Comment by leejo** [comment]
@abh - suggested changes made.

**Comment by abh** [comment]
Thanks!

**Pull request merged**